migration/20220428-namespace-overdrive-configuration.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+import logging
+import os
+import sys
+from typing import Optional
+
+from sqlalchemy import text
+from sqlalchemy.orm import Session
+
+bin_dir = os.path.split(__file__)[0]
+package_dir = os.path.join(bin_dir, "..")
+sys.path.append(os.path.abspath(package_dir))
+
+from core.model import production_session
+
+logging.basicConfig()
+logger = logging.getLogger()
+logger.setLevel("INFO")
+
+
+def delete_key(db: Session, integration: int, key: str) -> None:
+    delete_key = text(
+        """
+DELETE FROM configurationsettings
+WHERE key = :key
+AND external_integration_id = :id
+"""
+    )
+    db.execute(delete_key, {"id": f"{integration}", "key": key})
+
+
+def rename_key(db: Session, integration: int, old_key: str, new_key: str) -> None:
+    rename_key = text(
+        """
+UPDATE configurationsettings
+SET key = :new_key
+WHERE key = :old_key
+AND external_integration_id = :id
+    """
+    )
+    db.execute(
+        rename_key, {"id": f"{integration}", "old_key": old_key, "new_key": new_key}
+    )
+
+
+def update_key_value(db: Session, integration: int, old_key: str, new_key: str) -> int:
+    # Try to retrieve the values of any old or new keys.
+    # Note that the CM's configuration complicates things by
+    # having keys that may be present but with null values.
+    select_old_key = text(
+        """
+SELECT c.value FROM configurationsettings AS c
+  WHERE c.key = :key
+    AND c.external_integration_id = :id
+    """
+    )
+    result_old_key = db.execute(
+        select_old_key, {"id": f"{integration}", "key": old_key}
+    )
+
+    existing_old_key_present: bool = False
+    existing_old_key: Optional[str] = None
+    if result_old_key.rowcount > 0:
+        existing_old_key_present = True
+        existing_old_key = result_old_key.fetchone()[0]
+
+    select_new_key = text(
+        """
+SELECT c.value FROM configurationsettings AS c
+  WHERE c.key = :key
+    AND c.external_integration_id = :id
+    """
```

Comment:
Do you need to separate values for `select_old_key` and `select_new_key`? They look the same to me, though you would still need to use different params for `result_old_key` and `result_new_key`.